packages/generator/src/unique-name-finder.ts

```diff
@@ -0,0 +1,146 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import {
+  reservedObjectPrototypeKeywords,
+  reservedVdmKeywords
+} from './name-formatting-reserved-key-words';
+
+type Separator = '-' | '_';
+
+export class UniqueNameFinder {
+  private static readonly MAXIMUM_NUMBER_OF_SUFFIX = 1000;
+
+  public static getInstance(): UniqueNameFinder {
+    return new UniqueNameFinder();
+  }
```

Comment:
I was not sure why I don't like this static function, but finally it popped my mind - isn't that the typical wording when instantiating a singleton? This obviously is not a singleton. Also a static builder function like this makes sense if setting up an instance is more complex, here noting is even being passed to the constructor.
Which in turn I actually think should be the case. Setting the separator and already used names should be done in the constructor. I would propose an overloaded constructor for this.

Reply:
In Java I think the getInstance() is also used for non singleton cases. Actually I wanted to do a builder first where you could set the attributes, but then I thought this would even be less loved by the functional TS crowd. So I go for a constructor.